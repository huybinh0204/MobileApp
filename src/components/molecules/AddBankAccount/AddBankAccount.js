import { useTheme } from '@emotion/react';
import PropTypes from 'prop-types';
import React from 'react';
import { ActivityIndicator, TouchableOpacity, View } from 'react-native';
import PlaidLink from 'react-native-plaid-link-sdk';
import { IconSvg } from '_components/atoms';
import { ListItem } from '_components/molecules';
import { ICONS } from '_constants';
import strings from '_localization';
import { normalize } from '_utilities/screen';

const AddBankAccount = ({
  linkedAccount,
  token,
  showAccountInfo,
  onLinkAccountExit,
  onLinkAccountSuccess,
  loading,
}) => {
  const { colors } = useTheme();

  if (linkedAccount && !loading) {
    return (
      <TouchableOpacity onPress={showAccountInfo} testID="linkedAccountButton">
        <ListItem color={colors.beta900} icon={ICONS.addBankAccount} title={linkedAccount?.name}>
          <IconSvg icon={ICONS.arrowRight} height={normalize(25)} width={normalize(25)} />
        </ListItem>
      </TouchableOpacity>
    );
  }

  if (token && !loading) {
    return (
      <View testID="plaidLinkAccountButton">
        <PlaidLink
          onExit={onLinkAccountExit}
          onSuccess={onLinkAccountSuccess}
          tokenConfig={{ token }}
        >
          <ListItem title={strings.accounts.add} icon={ICONS.addBankAccount}>
            <IconSvg icon={ICONS.arrowRightActive} height={normalize(25)} width={normalize(25)} />
          </ListItem>
        </PlaidLink>
      </View>
    );
  }

  return <ActivityIndicator color={colors.alpha500} size="small" testID="addAccountLoading" />;
};

AddBankAccount.propTypes = {
  linkedAccount: PropTypes.oneOfType([PropTypes.bool, PropTypes.object]),
  token: PropTypes.string,
  showAccountInfo: PropTypes.func,
  onLinkAccountExit: PropTypes.func,
  onLinkAccountSuccess: PropTypes.func,
  loading: PropTypes.bool,
};

AddBankAccount.defaultProps = {
  linkedAccount: false,
  token: null,
  showAccountInfo: () => {},
  onLinkAccountExit: () => {},
  onLinkAccountSuccess: () => {},
  loading: false,
};

export default AddBankAccount;
