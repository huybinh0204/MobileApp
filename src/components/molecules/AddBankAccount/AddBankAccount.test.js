import { render } from '@testing-library/react-native';
import React from 'react';
import ComponentWithProviders from '_test/mocks/ComponentWithProviders';
import AddBankAccount from './AddBankAccount';

describe('AddBankAccount', () => {
  it('should match the snapshot', () => {
    const { toJSON } = render(
      <ComponentWithProviders
        component={AddBankAccount}
        componentProps={{
          linkedAccount: {
            id: '123',
            name: 'Plaid Test 1230',
          },
          loading: false,
        }}
      />
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should render plaid button', () => {
    const { getByTestId } = render(
      <ComponentWithProviders
        component={AddBankAccount}
        componentProps={{
          token: 'plaid-token',
          linkedAccount: null,
          loading: false,
        }}
      />
    );
    const plaidBtn = getByTestId('plaidLinkAccountButton');
    expect(plaidBtn).toBeTruthy();
  });

  it('should render the activity indicator', () => {
    const { getByTestId } = render(
      <ComponentWithProviders
        component={AddBankAccount}
        componentProps={{
          linkedAccount: null,
          loading: true,
        }}
      />
    );

    const addAccountLoading = getByTestId('addAccountLoading');
    expect(addAccountLoading).toBeTruthy();
  });

  it('should render the linked account info', () => {
    const { getByTestId } = render(
      <ComponentWithProviders
        component={AddBankAccount}
        componentProps={{
          token: null,
          linkedAccount: {
            id: '123',
            name: 'Plaid Test 1230',
          },
          loading: false,
        }}
      />
    );
    const linkedAccountButton = getByTestId('linkedAccountButton');
    expect(linkedAccountButton).toBeTruthy();
  });
});
