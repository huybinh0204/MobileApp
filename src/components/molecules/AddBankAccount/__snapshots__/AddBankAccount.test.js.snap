// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddBankAccount should match the snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                  "marginTop": 0,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        accessible={true}
                        collapsable={false}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "opacity": 1,
                          }
                        }
                        testID="linkedAccountButton"
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "alignSelf": "stretch",
                                "flexDirection": "row",
                                "justifyContent": "space-between",
                              },
                              undefined,
                            ]
                          }
                        >
                          <View
                            style={
                              Array [
                                Object {
                                  "alignItems": "center",
                                  "flex": 1,
                                  "flexDirection": "row",
                                },
                                undefined,
                              ]
                            }
                          >
                            <View
                              style={
                                Array [
                                  Object {
                                    "marginRight": 29,
                                  },
                                  undefined,
                                ]
                              }
                              testID="LeftIcon"
                            >
                              <SvgMock
                                height={54.5}
                                width={54.5}
                              />
                            </View>
                            <View
                              style={
                                Array [
                                  Object {
                                    "width": "100%",
                                  },
                                  undefined,
                                ]
                              }
                            >
                              <Text
                                allowFontScaling={false}
                                color="#1F1D25"
                                numberOfLines={1}
                                style={
                                  Array [
                                    Object {
                                      "color": "#1F1D25",
                                      "fontFamily": "Poppins-SemiBold",
                                      "fontSize": 25.5,
                                      "lineHeight": 35.5,
                                      "width": "80%",
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                Plaid Test 1230
                              </Text>
                            </View>
                          </View>
                          <SvgMock
                            height={45.5}
                            width={45.5}
                          />
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "height": 44,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
    </View>
  </RNCSafeAreaProvider>
</View>
`;
